### DISCUSSION TITLE
Document validation

### DISCUSSION THREAD

**Discussion by Ibrahim-Halil-Kuray** [ProxeusApp/community#32] [Q&A]
I have some questions regarding the logic we use for verifying PDF documents. I'm wondering what exactly the logic entails. Are the hash values assigned to these documents stored somewhere in our system, and then later compared to the hash value in the document for verification? Or is there no data stored in our system, and the comparison is only made with the hash values stored in the blockchain? Or is another method being used?

Additionally, there's another topic that I'm having difficulty understanding. When we change the network / test net of our application, will the documents we previously created still remain valid and verifiable, or will they become invalid?

Another question related to this topic is: Do individuals who want to use our system only to check the authenticity of a single document need to set up Metamask wallets and connect them to our application?

Thank you.

**Comment by loleg** [comment]
Thanks for these pertinent questions!

Hash values are not stored in the system, they are stored in the blockchain using a Smart Contract. Every time you upload a document for verification, the hash is generated a new for comparison.

This means that if you change the network of the application server, all the documents that have been created on the old network will no longer be verifiable. Therefore make sure to use a stable production network if possible for any long-term usage.

You do not need a Metamask wallet to check a document, only to create new ones, or sign documents that you have been invited to.
